components/script/dom/htmlformelement.rs

```diff
@@ -599,28 +646,35 @@ impl HTMLFormElement {
         let base = doc.base_url();
         // TODO: Handle browsing contexts (Step 4, 5)
         // Step 6
-        if submit_method_flag == SubmittedFrom::NotFromForm && !submitter.no_validate(self) {
-            if self.interactive_validation().is_err() {
-                // TODO: Implement event handlers on all form control elements
-                self.upcast::<EventTarget>().fire_event(atom!("invalid"));
+        if submit_method_flag == SubmittedFrom::NotFromForm {
```

Comment:
Please number the various Step 6 substeps, like you've already done with 6.4 below

Reply:
done, for the whole function (numbers were off, I suppose the spec changed at some point)